### PR TITLE
feature/data-scheme-manager

### PR DESCRIPTION
**Pull request by Timbogen**

### CODE REVIEW

File: server/src/util/test.util.ts [hunk 1]
```diff
@@ -54,4 +62,77 @@ export default class TestUtil {
...
+          RETURN l {. *} AS label`;
+
+        const params = {
+            labelName: l.name,
+            color: l.color,
+            attribs: JSON.stringify(l.attributes),
+        };
+
+        const resolveWrite = (res) => {
+            return res.records[0].get("label").name;
```
Comment (stefan-rink) [inline server/src/util/test.util.ts:116]: may remove the return an write it as an one-liner

**Review by stefan-rink — APPROVED** [review]

File: client/src/modules/schemes/modules/relation-editor/RelationEditor.vue [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+
+            <ConnectionView
+                v-for="(connection, index) in modifiedRelation.connections"
+                :key="`${connection.name}-${objectUUID(connection)}`"
+                v-model:from="connection.from"
+                v-model:to="connection.to"
+                @delete="deleteConnection(index)"
+            />
+
+            <!-- The save button --->
```
Comment (stefan-rink) [inline client/src/modules/schemes/modules/relation-editor/RelationEditor.vue:57]: The save-button will not be enabled if you just change the direction of an existing connection

**Review by stefan-rink — CHANGES_REQUESTED** [review]

File: server/src/relations/relations.controller.spec.ts [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+            attribs: JSON.stringify(l.attributes),
+        };
+
+        const resolveWrite = (res) => {
+            return res.records[0].get("label").name;
+        };
+        return neo4jService
+            .write(cypher, params, process.env.DB_TOOL)
+            .then(resolveWrite)
+            .catch(databaseUtil.catchDbError);
```
Comment (failipp) [inline server/src/relations/relations.controller.spec.ts:155]: Please use helper functions defined in test.util.ts

File: server/src/relations/relations.service.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            type: record.get("relation").type,
+            attributes: attributes,
+            from: record.get("relation").from,
+            to: record.get("relation").to,
+        } as Relation;
+
+        delete relation["attributes"]["relationId"];
+
+        return this.dataSchemeUtil.parseRecordByRelationType(relation);
+    }
```
Comment (failipp) [inline server/src/relations/relations.service.ts:80]: Use helper function from data-scheme.util.ts

File: server/src/util/data-scheme.util.ts [hunk 5]
```diff
@@ -1,24 +1,102 @@
...
+        const attributes = record.get("node");
+
+        const node = {
+            nodeId: record.get("node").nodeId,
+            name: record.get("node").name,
+            label: record.get("node").label,
+            attributes: attributes,
+        } as Node;
+
+        delete node["attributes"]["nodeId"];
```
Comment (failipp) [inline server/src/util/data-scheme.util.ts:66]: NodeId shouldn't be deleted from attributes, if it is specified in the data scheme, it should be transmitted.

File: server/src/relations/relations.service.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+        const relation = {
+            relationId: record.get("relation").relationId,
+            type: record.get("relation").type,
+            attributes: attributes,
+            from: record.get("relation").from,
+            to: record.get("relation").to,
+        } as Relation;
+
+        delete relation["attributes"]["relationId"];
```
Comment (failipp) [inline server/src/relations/relations.service.ts:77]: RelationId shouldn't be deleted from attributes, if it is specified in the data scheme, it should be transmitted.

**Review by MaxPeltzer — APPROVED** [review]
Reviewed relations.service queries. Everything fine.

**Review by failipp — APPROVED** [review]

**Review by stefan-rink — APPROVED** [review]

**Pull request merged**